src/states/elder/mod.rs

```diff
@@ -616,6 +639,25 @@ impl Elder {
         Ok(())
     }
 
+    fn add_to_proving_section_cache(
+        &mut self,
+        proving_secs: Vec<ProvingSection>,
+        sec_info: SectionInfo,
+    ) {
+        if self.chain.is_new_neighbour(&sec_info) {
+            self.set_of_proving_sections
```

Comment:
similar to filter, not sure if just checking leaf sec_info as valid neighbour is enough if the src_section is a couple hops away but proving_section contains new info about our valid neighbour.